client/client-multi/client-c/src/kaa/kaa_event.c

```diff
@@ -706,30 +714,45 @@ static kaa_error_t kaa_event_read_event(kaa_event_manager_t *self, kaa_platform_
         error = kaa_platform_message_read(reader, &event_data_size, sizeof(uint32_t));
         if (error) {
             KAA_LOG_ERROR(self->logger, error, "Failed to read event data size field");
-            return error;
+            goto end;
         }
         event_data_size = KAA_NTOHL(event_data_size);
     }
 
-    kaa_endpoint_id event_source;
+    event_source = (uint8_t *)KAA_MALLOC(KAA_ENDPOINT_ID_LENGTH);
```

Comment:
I doubt that there is a strong need to allocate this buffer every time.
I suggest to create the buffer once either in a static manner or maybe as the member of the event manager.